src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -230,9 +240,20 @@ private XElement GetPackageMappingsElementForSource(string packageSource)
                 {
                     pkgSrc.Add(new XElement("package", new XAttribute("pattern", packagePattern)));
                 }
-                else
+                else if (packageSource.Equals(PreviouslySourceBuiltSourceName))
                 {
-                    foreach (string version in previouslyBuiltPackages[packagePattern])
+                    foreach (string version in previouslySourceBuiltPackages[packagePattern])
```

Comment:
The previous and prebuild bodies are identical other than the source they iterate on.  Consider refactoring into a shared method.